src/events/views.py

```diff
@@ -262,6 +263,22 @@ def get_event(self):
             return None
         return event
 
+    def get_context_data(self, **kwargs):
+        community_track_event = None
+        try:
+            community_track_event = (
+                CommunityTrackEvent.objects
+                                .select_related('begin_time', 'end_time')
+                                .filter(talk=self.object).first()
+            )
+        except CommunityTrackEvent.DoesNotExist:
```

Comment:
try block is unnecessary because `first()` returns None by default. https://docs.djangoproject.com/en/3.0/ref/models/querysets/#first

Reply:
Oh yeah... I must have been distracted. I'll make the change here.

I kind of want to mimic what `get_event()` function does, but I noticed that the ProposedTalkEvent <-> TalkProposal is not a OneToOne mapping, but a ForeignKey instead, so there might be more than one event returned, and there's no code catching `MultipleObjectsReturned`. So I used this simple way, `.filter(...).first()`, and I guess I have already layout the try catch clause but changed my mind in the middle and forgot to remove the try catch.

Reply:
Or we can change the relation to one-to-one instead 🙂

Reply:
Will it break the database foreign key link if I change `ForeignKey` to `OneTnOneField`? I can experiment though :P but just curious if anyone know this in advance.

Reply:
The main source of breakage would be reverse relationships, since Django returns a [RelationManager for a ForeignKey’s reverse relationship](https://docs.djangoproject.com/en/3.1/ref/models/relations/), but [an object or None for one-to-one](https://docs.djangoproject.com/en/3.1/topics/db/queries/#one-to-one-relationships). If you don’t want to fix *any* errors, you can do `ForeignKey(..., unique=True)` instead, which would keep the reverse relationship to return a RelationManager that always contains at most one item.

Reply:
OK. I think I'll do the `unique=True` on current ForeignKey since it's a little bit too close to the conference and I don't want to cause massive possible breakage now. This should be carried over to the next year.